### PR TITLE
Add elm-todo-rest-api

### PR DESCRIPTION
**Pull request by andrewsuzuki**
The official elm-todomvc uses localStorage to persist state.

Then there's todomvc-elmfire which uses Firebase (and elm 0.16 signals).

Both of them are in single elm source files as well.

Since vanilla rest APIs are by far the most common type of web app persistence (and real-world elm apps should be much more modular), I decided to make another elm todo app with a fake json rest api.
The code is heavily-documented so it suits beginner users better than the other two as well.

Would love to hear any feedback!

[elm-todo-rest-api](https://github.com/andrewsuzuki/elm-todo-rest-api)

### CODE REVIEW

**Comment by decentral1se** [comment]
> The code is heavily-documented so it suits beginner users better than the other two as well.

Huzzah! Thank you and well done. Seeing the project structure is really helpful.

**Comment by isRuslan** [comment]
Thanks!

File: README.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ Inspired by the [awesome](#more-awesome) list thing. Feel free to <a href="https
 * [Vessel](https://github.com/slawrence/vessel) - A "tunnel" game written in Elm!
 * [Sliding Puzzle](https://github.com/moroshko/sliding-puzzle) - Configurable sliding puzzle game written in Elm.
 * [TodoMVC](https://github.com/evancz/elm-todomvc) - Proper implementation of the TodoMVC app.
+* [TodoMVC with JSON API](https://github.com/andrewsuzuki/elm-todo-rest-api) - Bare-bones, modular, heavilty-documented todo app with JSON API persistence.
```
Comment (decentral1se) [inline README.md:41]: Oh wait, there is a typo :bomb:

Comment (isRuslan) [inline README.md:41]: @lwm thanks, fixed in [b2cce6cbc0ff837838a8e9344c182a4b55a31e4f](https://github.com/isRuslan/awesome-elm/commit/b2cce6cbc0ff837838a8e9344c182a4b55a31e4f)

**Pull request merged**